### PR TITLE
Remove lsp-proxy, indexer, xlang-*

### PR DESCRIPTION
**Pull request by chrismwendt**
I roughly did the following:

- `grep -r "lsp-proxy|indexer|xlang" .` and remove occurrences
- `git ls-files | grep "lsp-proxy|indexer|xlang" | xargs rm`

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Test what is currently checked in to master by [installing](docs/install.md) Sou
    - In order to give yourself permissions to create roles on the cluster, run: `kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $YOUR_NAME@sourcegraph.com`
 1. Use the instructions in [configure.md](docs/configure.md) to:
    1. Add a repository (e.g. [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph))
-   1. [Add a language server](docs/configure.md#Configure-language-servers) (e.g. [Go](configure/xlang/go/README.md)), and test that code intelligence is working on the above repository
+   1. Enable a language extension (e.g. [Go](https://sourcegraph.com/extensions/chris/lang-go)), and test that code intelligence is working on the above repository
```
Comment (nicksnyder) [inline README.dev.md]: Were you planning on renaming this to sourcegraph/lang-go?

File: configure/prometheus/prom-metrics.md [hunk 2]
```diff
@@ -102,40 +102,16 @@ you can [import these dashboards directly from JSON](/docs/grafana/).
...
-
-    ```
-    prod:xlang_errors:ratio5m
-    ```
-
 1.  restarts, by service:
 
     ```
-    sum by (container)(increase(kube_pod_container_status_restarts{container=~\"(lsp|xlang).*\"}[10m]))
+    sum by (container)(increase(kube_pod_container_status_restarts{container=~\"(lsp).*\"}[10m]))
```
Comment (nicksnyder) [inline configure/prometheus/prom-metrics.md]: lsp?

File: configure/prometheus/prom-metrics.md [hunk 3]
```diff
@@ -102,40 +102,16 @@ you can [import these dashboards directly from JSON](/docs/grafana/).
...
 
     ```
-    max by (container_name)(task:container_cpu_usage_seconds_total:sum{container_name=~\"(lsp|xlang).*\"})
-    ```
-
-1.  max RAM, by service:
-
-    ```
-    max by (container_name)(container_memory_rss{container_name=~\"(xlang|lsp).*\"}) / 1024 / 1024 / 1024
+    max by (container_name)(task:container_cpu_usage_seconds_total:sum{container_name=~\"(lsp).*\"})
```
Comment (nicksnyder) [inline configure/prometheus/prom-metrics.md]: lsp?

File: docs/configure.md [hunk 4]
```diff
@@ -269,33 +269,14 @@ your fork is private **and** you are okay with storing secrets in it).
...
-- [OCaml](../configure/xlang/experimental/ocaml/README.md)
-- [R](../configure/xlang/experimental/r/README.md)
-- [Ruby](../configure/xlang/experimental/ruby/README.md)
-- [Rust](../configure/xlang/experimental/rust/README.md)
+Code intelligence is provided through [Sourcegraph extensions](https://docs.sourcegraph.com/extensions). Refer to the READMEs for each language for instructions about how to deploy and configure them:
+
+- [Go](https://sourcegraph.com/extensions/chris/lang-go)
+- [JavaScript/TypeScript](https://sourcegraph.com/extensions/sourcegraph/lang-typescript)
+- [Python](https://sourcegraph.com/extensions/sourcegraph/python)
+- Java (coming soon)
```
Comment (nicksnyder) [inline docs/configure.md]: minor: Is there some other landing page that we can link to so we won't have to remember to update these links?

For example, link to a search on the extension registry that would include all current and future language servers?

File: README.dev.md (hunk 1 above)
Comment (chrismwendt) [reply README.dev.md]: Yup, renamed to `sourcegraph/lang-go`

File: configure/prometheus/prom-metrics.md (hunk 3 above)
Comment (chrismwendt) [reply configure/prometheus/prom-metrics.md]: Removed

File: docs/configure.md (hunk 4 above)
Comment (chrismwendt) [reply docs/configure.md]: https://sourcegraph.com/extensions?query=lang works pretty well, but would miss `sourcegraph/python`

I'll at least replace `(coming soon)` with links to `?query=java`, for example.

**Pull request merged**